### ISSUE TITLE
[Bug] Analyzer reports migration to 4.1.10 problem

### ISSUE THREAD

**Issue by mieczkowski** [TheHive-Project/TheHive#2203]
### Request Type
Bug

### Work Environment

| Question                   | Answer                                |
| -------------------------- | ------------------------------------- |
| TheHive version / git hash | 4.1.10               |
| Package Type               | Docker |
| Database                   | Cassandra                |
| Index type                 | Elasticsearch                |
| Attachments storage        | Local                 |

### Problem Description
After migration from TheHive 3.5.1 to 4.1.10 everything was migrated but reports from cortex are not visible in observables list:

![1](https://user-images.githubusercontent.com/3045505/133458698-886ade2d-7ac0-4acd-9b0e-1f53fcddc702.png)

instead of (hive3):

![3](https://user-images.githubusercontent.com/3045505/133459723-019ddb32-5b13-4c1e-a073-86d7a97eabae.png)

and in observable details we have all analyzers available to run, but existing reports are under some hash name:

![2](https://user-images.githubusercontent.com/3045505/133458921-64f67a56-a079-4d0f-aa9b-0672518afd2c.png)

### Steps to Reproduce
1. Configure clean installation od hive4, configure cortex key and import cortex templates
2. Do migration...

**Comment by KRUXLEX** [comment]
@nadouani  @ferozsalam @vdebergue @2xyo 
Hi, we have same issue, please for fast response/fix for this.
We can't migrate without it

**Comment by mieczkowski** [comment]
Maybe here is the problem:

https://github.com/TheHive-Project/TheHive/blob/main/migration/src/main/scala/org/thp/thehive/migration/th3/Conversion.scala#L431

```
      workerId         <- (json \ "analyzerId").validate[String]
      workerName       <- (json \ "analyzerId").validate[String]
      workerDefinition <- (json \ "analyzerId").validate[String]
```

should be:

```
      workerId         <- (json \ "analyzerId").validate[String]
      workerName       <- (json \ "analyzerName").validate[String]
      workerDefinition <- (json \ "analyzerDefinition").validate[String]
```

I will try to test it

**Comment by mieczkowski** [comment]
Changing this json keys helped, reports are matched correctly, but report tags (taxonomies) are still unavailable. \

I think that they are not migrated because of:

```
(Json.parse(j) \ "full").asOpt[JsObject]
```

but I don't know how to fix this. Removing "full" have the same effect as with it.

**Comment by mieczkowski** [comment]
I've created pull request with analyzer report fix and with taxonomies migration.

**Comment by nadouani** [comment]
Hello, thanks for raising this issue, we will take a look and fix for 4.2
